src/distribute.js

```diff
@@ -0,0 +1,186 @@
+        case CATEGORY_CSV_SALE_PRIVATE:
+            throw 'CSV_CATEGORY_PRIVATE_SALE_HAS_NO_SOLIDITY_ENUM_VALUE';
+        case CATEGORY_CSV_SALE_PUBLIC:
+            throw 'CSV_CATEGORY_PUBLIC_SALE_HAS_NO_SOLIDITY_ENUM_VALUE';
+        default:
+            throw 'CSV_CATEGORY_UNKNOWN';
+    }
+};
+
+const transfer1PhtToEveryDistributionAddress = async (data) => {
+    data.forEach(async function (distribution) {
+        console.log(`Transferring 1 PHT to ${distribution.to}...`);
+
+        await web3.eth.sendTransaction({
+            from: WALLET_PROJECT,
+            to: distribution.to,
+            value: pht2Wei('1'),
+            gas: 21000
+        });
+    });
+
+    console.log("All distribution accounts were topped-up!");
+};
+
+csv()
+.fromFile(CSV_FILEPATH)
+.then(async (data) => {
+    const txReceipts = [];
+    const contract = initContract();
+
+    try {
+        await transfer1PhtToEveryDistributionAddress(data);
+    } catch (e) {
+        process.exit(1);
+    }
+
+    data.forEach(async function (distribution) {
+        const gasLimit = '300000';
+
+        const to = distribution.to;
+        const purchased = pht2Wei(distribution.purchased_pht);
+        const bonus = pht2Wei(distribution.bonus_pht);
+        const category = distribution.category;
+        const value = calculateDepositValue(purchased, bonus);
+
+        if (isProject(category)) {
+            logDeposit(WALLET_PROJECT, to, value, category);
+
+            const solidityCategory = csvCategoryToSolidityEnumValue(category);
+
+            contract.methods
+                .scheduleProjectVesting(to, solidityCategory)
+                .send({from: WALLET_PROJECT, value: value, gas: gasLimit})
+                .on('transactionHash', (hash) => {
+                    console.log(hash);
+                    txReceipts.push(hash);
+                })
+                .on('error', console.error);
+        } else if (isPrivateSale(category)) {
+            logDeposit(WALLET_SALE, to, value, category);
+
+            contract.methods
+                .schedulePrivateSaleVesting(to, bonus.toString())
+                .send({from: WALLET_SALE, value: value, gas: gasLimit})
+                .on('transactionHash', (hash) => {
+                    console.log(hash);
+                    txReceipts.push(hash);
+                })
+                .on('error', console.error);
+        } else if (isPublicSale(category)) {
+            logDeposit(WALLET_SALE, to, value, category);
+
+            contract.methods
+                .schedulePublicSaleVesting(to, bonus.toString())
+                .send({from: WALLET_SALE, value: value, gas: gasLimit})
+                .on('transactionHash', (hash) => {
+                    console.log(hash);
+                    txReceipts.push(hash);
+                })
+                .on('error', console.error);
+        } else {
+            throw 'CSV_CATEGORY_UNKNOWN'
+        }
+    });
+
+    let maxTimeoutSeconds = 15;
```

Comment:
[MAJOR] How did you estimate this `maxWait` time? What it happens in case the wait is exceeded and you didn't receipt all txReceipt? 

Keep in mind once the process exists you won't have more log outputs and every ongoing execution will be killed and you cannot know if the tx was sent or not.